src/shogun/base/SGObject.h

```diff
@@ -676,7 +676,10 @@ class CSGObject
 	void watch_param(
 		const std::string& name, T* value,
 		AnyParameterProperties properties = AnyParameterProperties(
-		    "Unknown parameter", MS_NOT_AVAILABLE, GRADIENT_NOT_AVAILABLE))
+			"Unknown parameter",
+			AnyParameterProperties::HYPERPARAMETER |
```

Comment:
instead of doing this here, I think it might be better to offer a constructor that does not take any properties, and then set the default values inside there. We need to discuss clever default values as well btw